packages/database/src/modelClasses/Entity.js

```diff
@@ -374,6 +374,38 @@ export class EntityModel extends DatabaseModel {
     );
   }
 
+  async getEntityAnswerWithEntityName(dataSourceEntities, questionCode) {
```

Comment:
I don't think this method should live in the model, since its concern is very specific. It's better if we move it to a utility function.

Models should be focused on described the basic underlying "truths" about objects stored in the DB, and their relationships (the `model` of our application). The details that sit on top of that are part of the `business` logic, which can change more often and is more related to a specific client's needs (eg connecting entities to survey responses via `Entity`/`PrimaryEntity` questions)